common/components/BalanceSidebar/TokenBalances.jsx

```diff
@@ -0,0 +1,78 @@
+// @flow
+import React from 'react';
+import translate from 'translations';
+import TokenRow from './TokenRow';
+import AddCustomTokenForm from './AddCustomTokenForm';
+import type { TokenBalance } from 'selectors/wallet';
+import type { Token } from 'config/data';
+
+export default class TokenBalances extends React.Component {
+  props: {
+    tokens: TokenBalance[],
+    onAddCustomToken: (token: Token) => any,
+    onRemoveCustomToken: (symbol: string) => any
+  };
+  state = {
+    showAllTokens: false,
+    showCustomTokenForm: false
+  };
+
+  render() {
+    const { tokens } = this.props;
+    return (
+      <section className="token-balances">
+        <h5>{translate('sidebar_TokenBal')}</h5>
+        <table className="account-info">
+          <tbody>
+            {tokens
+              .filter(token => !token.balance.eq(0) || token.custom || this.state.showAllTokens)
+              .map(token =>
+                <TokenRow
+                  key={`${token.symbol}`}
+                  balance={token.balance}
+                  symbol={token.symbol}
+                  custom={token.custom}
+                  onRemove={this.props.onRemoveCustomToken}
+                />
+              )}
+          </tbody>
+        </table>
+        {!this.state.showAllTokens &&
+          <a className="btn btn-default btn-sm" onClick={this.toggleShowAllTokens}>
+            Show All Tokens
+          </a>}
+        {this.state.showAllTokens &&
+          <a className="btn btn-default btn-sm" onClick={this.toggleShowAllTokens}>
+            Hide Tokens
+          </a>}{' '}
```

Comment:
You can simplify lines 40 - 47! 😄 

The text of "Show All Tokens" and "Hide Tokens" are the only things changing, so just keep the markup in place and have the if-statement be for those strings only. Something like:

`{!this.state.showAllTokens ? 'Show All Tokens' : 'Hide Tokens'}`